### PR TITLE
feat: add fieldNames option in instance#getMetadata()

### PR DESCRIPTION
**Pull request by AVaksman**
- [x] Ensure the tests and linter pass
- [x] Code coverage does not decrease (if any source code was changed)
- [x] Appropriate docs were updated (if necessary)

Enable user to pass a list of `Instance` field names to be returned via `getMetadata()` method
[proto source](https://github.com/googleapis/googleapis/blob/master/google/spanner/admin/instance/v1/spanner_instance_admin.proto#L472)

### CODE REVIEW

File: src/instance.ts [hunk 1]
```diff
@@ -697,6 +715,9 @@ class Instance extends common.GrpcServiceObject {
    * @see {@link v1.InstanceAdminClient#getInstance}
    * @see [GetInstance API Documentation](https://cloud.google.com/spanner/docs/reference/rpc/google.spanner.admin.instance.v1#google.spanner.admin.instance.v1.InstanceAdmin.GetInstance)
    *
+   * @param {GetInstanceMetadataOptions} [options] Configuration object
+   * @param {string | string[]} [options.fieldNames] A list of `Instance` field
+   *     names to be requested.
```
Comment (laljikanjareeya) [inline src/instance.ts]: it's good to have example of `fieldNames`.

File: test/instance.ts [hunk 2]
```diff
@@ -498,6 +532,34 @@ describe('Instance', () => {
...
+          fieldMask: {
+            paths: fieldNames,
+          },
+          name: instance.formattedName_,
+        });
+        done();
+      };
+      instance.get({fieldNames}, assert.ifError);
+    });
+
```
Comment (laljikanjareeya) [inline test/instance.ts]: can we add unit test case for nested type as mention in below docs:
https://github.com/googleapis/nodejs-spanner/blob/1c0cf27b6c502fe7cc03f9b4c9191d746761b8b6/protos/google/protobuf/field_mask.proto#L42-L174

Comment (AVaksman) [reply test/instance.ts]: I believe there is no case of nested types for `FieldMask` in `getMetadata()` for `Instance`
Available values for `fieldNames` array are:
`'name'`, `'displayName'`, `'endpointUris'`, `'labels'`, `'config'`, `'nodeCount'`, `'state'`

File: src/instance.ts (hunk 1 above)
Comment (AVaksman) [reply src/instance.ts]: Added some samples as well as a list of eligible values that could be passed as `fieldNames`

**Comment by AVaksman** [comment]
While we at it, how about utilize `FieldName` functionality with `Instance#exists()` method.
Perhaps pass `name` into `getMetadata` inside the `exists` [method](https://github.com/AVaksman/nodejs-spanner/blob/instance_getMeta_fieldMaks/src/instance.ts#L477) as optimization effort?

File: src/instance.ts [hunk 3]
```diff
@@ -83,7 +86,9 @@ export type SetInstanceMetadataCallback = ResourceCallback<
   GaxOperation,
   IOperation
 >;
-export interface GetInstanceConfig extends GetConfig {}
+export interface GetInstanceConfig
+  extends GetConfig,
+    GetInstanceMetadataOptions {}
```
Comment (hengfengli) [inline src/instance.ts]: nits: can we put `GetInstanceMetadataOptions {}` into previous line? like: 

```typescript
export interface GetInstanceConfig
  extends GetConfig, GetInstanceMetadataOptions {}
```

File: src/instance.ts [hunk 4]
```diff
@@ -542,7 +550,13 @@ class Instance extends common.GrpcServiceObject {
         ? optionsOrCallback
         : ({} as GetInstanceConfig);
 
-    this.getMetadata((err, metadata) => {
+    const getMetadataOptions: GetInstanceMetadataOptions = new Object(null);
+    if (options.fieldNames) {
+      getMetadataOptions.fieldNames = options[`fieldNames`];
```
Comment (hengfengli) [inline src/instance.ts]: nits: use single quote? `options['fieldNames']`.

File: src/instance.ts (hunk 3 above)
Comment (AVaksman) [reply src/instance.ts]: I tried, but unfortunately linter is complaining about it.
`npm run fix` breaks it back in two lines.

File: src/instance.ts (hunk 4 above)
Comment (AVaksman) [reply src/instance.ts]: done, thanks

**Pull request merged**